staging/src/k8s.io/client-go/util/csaupgrade/upgrade.go

```diff
@@ -50,10 +50,13 @@ var csaAnnotationFieldSet = fieldpath.NewSet(fieldpath.MakePathOrDie("metadata",
 // obj - Target of the operation which has been managed with CSA in the past
 // csaManagerName - Name of FieldManager formerly used for `Update` operations
 // ssaManagerName - Name of FieldManager formerly used for `Apply` operations
+// keepLastApplied - Controls whether the `kubectl.kubernetes.io/last-applied-configuration` annotation
+// is deleted from the object and from the manager's ownership
 func UpgradeManagedFields(
 	obj runtime.Object,
 	csaManagerName string,
 	ssaManagerName string,
+	keepLastApplied bool,
```

Comment:
As discussed in SIG, we would like to retain the last applied annotation for now. This is OK, since server-side-apply has logic to keep it up to date from the server-side if it exists.

Reply:
This methods isn't used yet, it's still brand new, but we have people using server-side apply outside the context of kubectl, and they may want to get rid of the last applied annotation.

That being said, I think if someone wants to remove the lastApplied annotation, it'd be simpler to have a `removeLastApplied()` (pretty sure it already exists in some form), rather than move the burden of specifying that parameter to everyone.